src/lib/libdll/dllscan.c

```diff
@@ -355,7 +355,7 @@ Dllent_t *dllsread(register Dllscan_t *scan) {
             if (scan->flags & (DLL_MATCH_NAME | DLL_MATCH_VERSION)) {
                 sfstrseek(scan->tmp, scan->off, SEEK_SET);
                 if (!(t = sfstruse(scan->tmp))) return 0;
-                if ((scan->fts = fts_open((char **)t, FTS_LOGICAL,
+                if ((scan->fts = fts_open((char **)&t, FTS_LOGICAL,
```

Comment:
This is the cause of the SIGSEGV. The first arg to `fts_open()` is a NULL terminated list of strings (like main()'s `argv` parameter). You can't just take a pointer to the pointer to the string. You have to construct a two element array where the first element is `t` and the second is `NULL`.

Reply:
And, of course, you didn't introduce this bug, @siteshwar, since it looks like it was probably a day one bug.

Reply:
Changing line 358 to this fixes the SIGSEGV:

```
char *argv[2] = {t, NULL};
if ((scan->fts = fts_open(argv, FTS_LOGICAL,
```

Reply:
And it also fixes the failure in the `exit` unit test.

Reply:
> And, of course, you didn't introduce this bug, @siteshwar, since it looks like it was probably a day one bug.

libast version of `fts_open()` supported this non-standard flag:

```
#define FTS_ONEPATH     0x00000020      /* pathnames arg is one const char*     */
```

so this bug got introduced when we switched to standard ```fts_open()``` function.